fullstack-cert/python-projects/workshop-media-catalogue/main.py

```diff
@@ -0,0 +1,120 @@
+class MediaError(Exception):
+    """Custom exception for media-related errors."""
+
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
+
+
+class Movie:
+    """Parent class representing a movie."""
+
+    def __init__(self, title, year, director, duration):
+        if not title.strip():
+            raise ValueError("Title cannot be empty")
+        if year < 1888:  # First movie was made in 1888
+            raise ValueError("Year must be 1888 or later")
+        if duration <= 0:
+            raise ValueError("Duration must be positive")
+
```

Comment:
```suggestion
    if not director or not director.strip(): 
        raise ValueError('Director cannot be empty')
```
you could also validate the Director